core/modules/server/authenticators/basic.js

```diff
@@ -60,35 +61,290 @@ Returns true if the request is authenticated and assigns the "authenticatedUsern
 Returns false if the request couldn't be authenticated having sent an appropriate response to the browser
 */
 BasicAuthenticator.prototype.authenticateRequest = function(request,response,state) {
```

Comment:
As you noted, you should add a new authenticators/cookie.js file. Things should just work if you implement the methods correctly, but we may need to tweak the logic in server.js